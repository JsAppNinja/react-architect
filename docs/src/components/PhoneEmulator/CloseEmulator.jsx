import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Button from 'react-md/lib/Buttons/Button';

export default class CloseEmulator extends PureComponent {
  static propTypes = {
    onClick: PropTypes.func,
  };

  static contextTypes = {
    hideDemo: PropTypes.func.isRequired,
  };

  handleClick = (e) => {
    if (this.props.onClick) {
      this.props.onClick(e);
    }

    this.context.hideDemo(e);
  };

  render() {
    return <Button {...this.props} onClick={this.handleClick} />;
  }
}
