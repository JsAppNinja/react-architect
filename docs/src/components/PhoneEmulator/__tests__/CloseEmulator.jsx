/* eslint-env jest */
import React from 'react';
import PropTypes from 'prop-types';
import { mount, render } from 'enzyme';

import CloseEmulator from '../CloseEmulator';

class ContextProvider extends React.Component {
  static propTypes = {
    hideDemo: PropTypes.func.isRequired,
    children: PropTypes.node.isRequired,
  };

  static defaultProps = {
    hideDemo: jest.fn(),
  };

  static childContextTypes = {
    hideDemo: PropTypes.func.isRequired,
  };

  getChildContext() {
    return { hideDemo: this.props.hideDemo };
  }

  render() {
    return this.props.children;
  }
}

describe('CloseEmulator', () => {
  it('should render correctly', () => {
    const tree1 = render(
      <ContextProvider>
        <CloseEmulator icon>close</CloseEmulator>
      </ContextProvider>
    );
    const tree2 = render(
      <ContextProvider>
        <CloseEmulator raised primary>Hello, World!</CloseEmulator>
      </ContextProvider>
    );
    expect(tree1).toMatchSnapshot();
    expect(tree2).toMatchSnapshot();
  });

  it('should correctly call the hideDemo context function and onClick', () => {
    const hideDemo = jest.fn();
    const onClick = jest.fn();

    const button = mount(
      <ContextProvider hideDemo={hideDemo}>
        <CloseEmulator icon onClick={onClick}>close</CloseEmulator>
      </ContextProvider>
    );

    button.simulate('click');
    expect(onClick).toBeCalled();
    expect(hideDemo).toBeCalled();
  });
});
