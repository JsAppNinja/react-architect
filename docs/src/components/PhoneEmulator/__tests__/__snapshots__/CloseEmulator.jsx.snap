// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseEmulator should render correctly 1`] = `
<button
  class="md-btn md-btn--icon md-pointer--hover md-inline-block"
  type="button"
>
  <div
    class="md-ink-container"
  />
  <i
    class="md-icon material-icons md-text--inherit"
  >
    close
  </i>
</button>
`;

exports[`CloseEmulator should render correctly 2`] = `
<button
  class="md-btn md-btn--raised md-btn--text md-pointer--hover md-background--primary md-background--primary-hover md-inline-block"
  type="button"
>
  <div
    class="md-ink-container"
  />
  Hello, World!
</button>
`;
